### DISCUSSION TITLE
14. Are contributors to open source exposing themselves to any liability of their solutions?

### DISCUSSION THREAD

**Discussion by MichaelRimler** [phuse-org/OSTCDA#14] [General]
Are contributors to open source exposing themselves to any liability of their solutions?

**Comment by KatjaGlassConsulting** [comment]
A core point of (most) Open Source licenses is that they specifically mention that there is no liability and warranty.

**Reply by MichaelRimler** [reply]
@KatjaGlassConsulting As an example, from the Apache 2 license in [admiral](https://github.com/pharmaverse/admiral/blob/devel/LICENSE.md):

7. Disclaimer of Warranty
Unless required by applicable law or agreed to in writing, Licensor provides the Work (and each Contributor provides its Contributions) on an “AS IS” BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied, including, without limitation, any warranties or conditions of TITLE, NON-INFRINGEMENT, MERCHANTABILITY, or FITNESS FOR A PARTICULAR PURPOSE. You are solely responsible for determining the appropriateness of using or redistributing the Work and assume any risks associated with Your exercise of permissions under this License.

8. Limitation of Liability
In no event and under no legal theory, whether in tort (including negligence), contract, or otherwise, unless required by applicable law (such as deliberate and grossly negligent acts) or agreed to in writing, shall any Contributor be liable to You for damages, including any direct, indirect, special, incidental, or consequential damages of any character arising as a result of this License or out of the use or inability to use the Work (including but not limited to damages for loss of goodwill, work stoppage, computer failure or malfunction, or any and all other commercial damages or losses), even if such Contributor has been advised of the possibility of such damages.

9. Accepting Warranty or Additional Liability
While redistributing the Work or Derivative Works thereof, You may choose to offer, and charge a fee for, acceptance of support, warranty, indemnity, or other liability obligations and/or rights consistent with this License. However, in accepting such obligations, You may act only on Your own behalf and on Your sole responsibility, not on behalf of any other Contributor, and only if You agree to indemnify, defend, and hold each Contributor harmless for any liability incurred by, or claims asserted against, such Contributor by reason of your accepting any such warranty or additional liability.

**Comment by andyofsmeg** [comment]
In any case, the sponsor is always responsible for the validation of anything they use.  This includes closed source.  I like the way the EMA summarises it in [EMA’s Notice on validation and qualification of software tools used in clinical trials](https://www.ema.europa.eu/en/documents/regulatory-procedural-guideline/notice-sponsors-validation-and-qualification-computerised-systems-used-clinical-trials_en.pdf),  _"The sponsor is ultimately responsible for the validation of the computerised system and for providing adequate documented evidence on the validation process.  [...] The sponsor may rely on qualification documentation provided by the vendor, if the qualification activities performed by the vendor have been assessed as adequate. However, the sponsor may also have to perform additional qualification (and validation) activities based on a documented risk assessment"_
